src/compiler/transformers/ts.ts

```diff
@@ -386,7 +379,7 @@ namespace ts {
                     return visitEnumDeclaration(<EnumDeclaration>node);
 
                 case SyntaxKind.AwaitExpression:
-                    // TypeScript 'await' expressions must be transformed.
+                    // Typescript ES2017 async/await are handled by ES2017 transformer

```

Comment:
I don't think you will even need this `case` if you mark async/await as ES2017 only